packages/web-app-files/tests/unit/components/SideBar/SideBar.spec.js

```diff
@@ -0,0 +1,256 @@
+      enabled: true,
+      icon: 'slideshow',
+      component: getSideBarComponent('Actions')
+    }
+  },
+  () => {
+    return {
+      app: 'sharing-item',
+      enabled: true,
+      icon: 'group',
+      component: getSideBarComponent('People')
+    }
+  }
+]
+
+function getWrapper(
+  route,
+  { filename = 'testfile', extension, type = 'file', publicLink = false }
+) {
+  const mountStubs = { ...stubs, 'oc-button': false }
+  const mountOptions = {
+    localVue,
+    store: createStore({}, filename, extension, type),
+    stubs: mountStubs,
+    mocks: {
+      $route: route,
+      publicPage: () => publicLink
+    },
+    directives: {
+      'click-outside': jest.fn()
+    }
+  }
+  return mount(SideBar, mountOptions)
+}
+
+const selectors = {
+  fileInfo: '.file_info .file_info__body',
+  panelHeader: '.sidebar-panel__header .header__title',
+  panelNavigation: '.sidebar-panel__navigation',
+  panelNavigationButton: '.sidebar-panel__navigation button',
+  panelBody: '.sidebar-panel__body',
+  actionButton: '#sidebar-panel-actions-item-select',
+  peopleButton: '#sidebar-panel-sharing-item-select'
+}
+
+const panels = {
+  Details: {
+    selector: '#sidebar-panel-details-item',
+    default: true
+  },
+  Actions: {
+    selector: '#sidebar-panel-actions-item'
+  },
+  People: {
+    selector: '#sidebar-panel-sharing-item'
+  }
+}
+
+describe('SideBar', () => {
+  describe('when user on personal route', () => {
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('should show panel buttons', () => {
+      const expectedPanels = ['Details', 'Actions', 'People']
+
+      for (const idx in expectedPanels) {
+        const wrapper = getWrapper(filesPersonalRoute, {
+          filename: 'testfile',
+          extension: 'jpg'
+        })
+        jest.clearAllMocks()
+
+        const panel = wrapper.find(panels[expectedPanels[idx]].selector)
+        expect(panel.exists()).toBeTruthy()
+        const fileInfo = panel.find(selectors.fileInfo)
+        expect(fileInfo.exists()).toBeTruthy()
+
+        const body = panel.find(selectors.panelBody)
+        expect(body.exists()).toBeTruthy()
+
+        const navigation = panel.find(selectors.panelNavigation)
+        if (panels[expectedPanels[idx]].default) {
+          expect(body.text()).toContain(expectedPanels[idx] + ' content')
+
+          expect(navigation.exists()).toBeTruthy()
+
+          const navigationButtons = panel.findAll(selectors.panelNavigationButton)
+          expect(navigationButtons.length).toBe(2)
+          for (let i = 0; i < 2; i++) {
+            if (expectedPanels[i] === 'Details') {
+              continue
+            }
+            const buttonName = navigationButtons.at(i).text()
+            expect(expectedPanels).toContain(buttonName)
+          }
+        } else {
+          expect(body.text()).toBe('')
+          expect(navigation.exists()).toBeFalsy()
+        }
+      }
+    })
+
+    it('Opening a panel should work', async () => {
```

Comment:
```suggestion
    it('should be able to open a panel', async () => {
```
just a suggestion.. but it would be nice to start with `it(should ...`